revokePermission.ps1

```diff
@@ -0,0 +1,283 @@
+                    IsError = $true
+                })
+
+            # Throw terminal error
+            throw
+        }
+    }
+
+    # Revoke permission
+    try {
+        Write-Verbose "Revoking permission to $($pRef.Name) ($($pRef.id)) for $($currentAccount.userName) ($($currentAccount.id))"
+
+        # Create permission body
+        $permissionBody = [PSCustomObject]@{
+            schemas    = "urn:ietf:params:scim:schemas:core:2.0:Group"
+            id         = $pRef.id
+            operations = @(
+                @{
+                    op    = "remove"
+                    path  = "members"
+                    value = @(
+                        @{
+                            value   = $currentAccount.id
+                            display = $currentAccount.userName
+                        }
+                    )
+                }
+            )
+        }
+
+        $body = ($permissionBody | ConvertTo-Json -Depth 10)
+        $splatWebRequest = @{
+            Uri             = "$baseUrl/scim/groups/$($pRef.Id)"
+            Headers         = $headers
+            Method          = 'PATCH'
+            Body            = ([System.Text.Encoding]::UTF8.GetBytes($body))
+            ContentType     = "application/json;charset=utf-8"
+            UseBasicParsing = $true
+        }
+
+        if (-Not($actionContext.DryRun -eq $true)) {
+            Write-Verbose "Revoking permission: [$($pRef.Name) ($($pRef.id))] to account: [$($currentAccount.userName) ($($currentAccount.id))]"
+
+            $revokePermission = Invoke-RestMethod @splatWebRequest -Verbose:$false
+
+            $outputContext.AuditLogs.Add([PSCustomObject]@{
+                    # Action  = "" # Optional
+                    Message = "Successfully revoked permission: [$($pRef.Name) ($($pRef.id))] to account: [$($currentAccount.userName) ($($currentAccount.id))]"
+                    IsError = $false
+                })
+        }
+        else {
+            Write-Warning "DryRun: Would revoke permission: [$($pRef.Name) ($($pRef.id))] to account: [$($currentAccount.userName) ($($currentAccount.id))]"
+        }
+    }
+    catch {
+        $ex = $PSItem
+        $errorMessage = Get-ErrorMessage -ErrorObject $ex
+
+        Write-Verbose "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
+        Write-Verbose "URI: $($splatWebRequest.Uri)"
+        Write-Verbose "Body: $($body)"
+
+        $outputContext.AuditLogs.Add([PSCustomObject]@{
+                # Action  = "" # Optional
+                Message = "Error revoking permission: [$($pRef.Name) ($($pRef.id))] to account: [$($currentAccount.userName) ($($currentAccount.id))]. Error Message: $($errorMessage.AuditErrorMessage)"
+                IsError = $true
+            })
+
+        # Throw terminal error
+        throw
+    }
+}
+catch {
+    $ex = $PSItem
```

Comment:
does this catch belong in the auditlogs?

Reply:
No, only used in dryrun